src/libcore/num/mod.rs

```diff
@@ -5250,3 +5250,71 @@ impl_from! { u32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0"
 
 // Float -> Float
 impl_from! { f32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0")] }
+
+// Conversion traits for non-zero integer types
+macro_rules! nzint_impl_from {
+    ($Small: ty, $Large: ty, #[$attr:meta], $doc: expr) => {
+        #[$attr]
+        #[doc = $doc]
+        impl From<$Small> for $Large {
+            #[inline]
+            fn from(small: $Small) -> $Large {
+                // SAFETY: input type guarantees the value is non-zero
+                unsafe {
+                    <$Large>::new_unchecked(small.get().into())
+                }
+            }
+        }
+    };
+    ($Small: ty, $Large: ty, #[$attr:meta]) => {
+        nzint_impl_from!($Small,
+                   $Large,
+                   #[$attr],
+                   concat!("Converts `",
+                           stringify!($Small),
+                           "` to `",
+                           stringify!($Large),
+                           "` losslessly."));
+    }
+}
+
+// Non-zero Unsigned -> Non-zero Unsigned
+nzint_impl_from! { NonZeroU8, NonZeroU16, #[unstable(feature = "nz_int_conv", issue = "66196")] }
```

Comment:
The "issue" is not a tracking issue. You need to open a new one. Take a look at https://github.com/rust-lang/rust/issues/66151 for example how to create one

Reply:
Aren't trait impls insta-stable? If so, I am not sure if tracking issue makes sense.

Reply:
@newpavlov you're right. Insta stability is something I hadn't in mind, but peter already fixed that ;)

Reply:
@newpavlov you're right. Insta stability is something I hadn't in mind, but peter already fixed that ;)